### PR TITLE
show an error if not collaborator

### PR DESCRIPTION
**Pull request by timothyclemansinsea**
For #1236

### CODE REVIEW

File: src/smc-webapp/project_files.cjsx [hunk 1]
```diff
@@ -1994,14 +1994,24 @@ exports.ProjectFiles = rclass ({name}) ->
...
-        if @props.redux.getStore('account').is_logged_in()
-            <ErrorDisplay title="Directory is not public" error={"You are trying to access a non public project that you are not a collaborator on. You need to ask a collaborator of the project to add you."} />
+    render_not_collaborator_error: ->
+        public_view = @props.get_my_group(@props.project_id) == 'public'
+        if public_view
+            if @props.redux.getStore('account').is_logged_in()
+                <ErrorDisplay title="Showing only public files" error={"You are trying to access a project that you are not a collaborator on. To view non-public files or edit files in this project you need to ask a collaborator of the project to add you."} />
+            else
+                <div>
+                    <ErrorDisplay title="Showing only public files" error={"You are not logged in. To view non-public files or edit files in this project you need to ask a collaborator of the project to add you."} />
```
Comment (haraldschilly) [inline src/smc-webapp/project_files.cjsx]: hmm, I'm probably confused, but this case here should include an emphasis on having to log in, just like the else case below (i.e. "If you are collaborator on this project you need to log in first.") -- or am I missing something?

**Comment by williamstein** [comment]
Changing tag to needs work, since Harald reviewed this and @timothyclemansinsea hasn't responded to that review.

**Comment by timothyclemansinsea** [comment]
@williamstein I did respond to him 
image

**Comment by williamstein** [comment]
Weird -- I do NOT see your response:

screen shot 2016-11-27 at 3 37 33 pm

**Comment by williamstein** [comment]
Oh -- @timothyclemansinsea in your screenshot it explicitly shows "Pending" very clearly.  You never submitted your response.

File: src/smc-webapp/project_files.cjsx [hunk 2]
```diff
@@ -2153,4 +2163,5 @@ exports.ProjectFiles = rclass ({name}) ->
             {@render_paging_buttons(Math.ceil(listing.length / file_listing_page_size)) if listing?}
             {@render_file_listing(visible_listing, file_map, error, project_state, public_view)}
             {@render_paging_buttons(Math.ceil(listing.length / file_listing_page_size)) if listing?}
+            <div>{@render_not_collaborator_error() if public_view}</div>
```
Comment (johnjeng) [inline src/smc-webapp/project_files.cjsx]: What is the div surrounding this for?

File: src/smc-webapp/project_files.cjsx [hunk 3]
```diff
@@ -1994,14 +1994,24 @@ exports.ProjectFiles = rclass ({name}) ->
                 style   = {error_style}
                 onClose = {=>@props.actions.setState(error:'')} />
 
-    render_not_public_error: ->
-        if @props.redux.getStore('account').is_logged_in()
-            <ErrorDisplay title="Directory is not public" error={"You are trying to access a non public project that you are not a collaborator on. You need to ask a collaborator of the project to add you."} />
+    render_not_collaborator_error: ->
```
Comment (johnjeng) [inline src/smc-webapp/project_files.cjsx]: Name this something like `render_access_error` since it covers publicity and collaborator status.

**Comment by timothyclemansinsea** [comment]
@johnjeng just checking that UI comment is for this PR and not the other
one.

**Pull request merged**